### PR TITLE
DOCSP-18552 page sections improvements

### PR DESCRIPTION
**Pull request by kyuan-mongodb**
## Pull Request Info

Improved each landing page section by:
- adding each section of the API documentation
- putting the learning material + forums at the bottom of the page

Note: The leafy guy wasn't used since the visual design team asked us not to.

Also, I updated the fundamentals drawers page (to the landing page's fundamentals section) since I forgot to in my previous PR.

### Issue JIRA link:
https://jira.mongodb.org/browse/DOCSP-18552

### Snooty build log:
https://workerpool-boxgs.mongodbstitch.com/pages/job.html?jobId=6143a97ac608bec57fdfcdb0

### Docs staging link (requires sign-in on MongoDB Corp SSO):
https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-18552-PageSectionsImprovements/

https://docs-mongodbcom-staging.corp.mongodb.com/java/docsworker-xlarge/DOCSP-18552-PageSectionsImprovements/fundamentals/

### Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Does it render on staging correctly?
- [x] Are all the links working?
- [x] Are the staging and workerpool job links in the PR description updated?

### If your page documents a concept, does it meet the following criteria?

- [ ] Target the [Jasmin persona](https://drive.google.com/file/d/14FbBOLCVxwSP6M9BK4Nz1Ir9tzxT8_02/view)
- [ ] Target the [Lucas persona](https://drive.google.com/file/d/1J2vqJxo7ldv7OP_obA9Q-avf0o_ju4Lk/view)

### CODE REVIEW

File: source/fundamentals.txt [hunk 1]
```diff
@@ -25,20 +25,20 @@ Fundamentals
    /fundamentals/csfle
    /fundamentals/time-series
 
-The Fundamentals section contains detailed information to help you
-expand your knowledge on how to use MongoDB with the Java driver. We
-divide the information in this section into the following topics: 
+The Fundamentals section is where you will find coverage of
```
Comment (ccho-mongodb) [inline source/fundamentals.txt:28]: [Reminder](https://docs.mongodb.com/meta/style-guide/writing/use-present-tense/
): avoid "will" (future tense) and use present tense instead.

File: source/fundamentals.txt [hunk 2]
```diff
@@ -25,20 +25,20 @@ Fundamentals
...
+- :doc:`Read from and Write to MongoDB </fundamentals/crud>`
 - :doc:`Simplify your Code with Builders </fundamentals/builders>`
-- :doc:`Perform Aggregations </fundamentals/aggregation>`
-- :doc:`Construct Indexes </fundamentals/indexes>`
-- :doc:`Specify Collations </fundamentals/collations>`
-- :doc:`Record Events in the Driver </fundamentals/logging>`
-- :doc:`Use Driver Events in your Code </fundamentals/monitoring>`
-- :doc:`Store and Retrieve Files in MongoDB </fundamentals/gridfs>`
+- :doc:`Transform your Data </fundamentals/aggregation>`
+- :doc:`Create Indexes to Efficiently Execute Queries </fundamentals/indexes>`
```
Comment (ccho-mongodb) [inline source/fundamentals.txt:38]: Nitpick: I think this could be easier for a wider audience to understand:

```suggestion
- :doc:`Create Indexes to Speed Up Queries </fundamentals/indexes>`
```

File: source/index.txt [hunk 3]
```diff
@@ -95,11 +100,33 @@ What's New
 The :doc:`What's New </whats-new>` section lists new features and
 changes in each version.
 
-Additional Learning
--------------------
+Resources
```
Comment (ccho-mongodb) [inline source/index.txt:103]: I thought you wanted this to be called "Learn" in the team PR review session which the team agreed with.
I think "Resources" omits the objective we want to emphasize ("learn").

It's also unclear what it refers to since it's used in the top level description as well as the Developer Hub one; this doesn't help the reader understand what a resource is and what the sections contain.

File: source/index.txt [hunk 4]
```diff
@@ -95,11 +100,33 @@ What's New
...
+the MongoDB Java driver.
+
+Developer Hub
+~~~~~~~~~~~~~
+
+The Developer Hub contains content, resources, community, and social
+engagement for developers.
+
+To view articles on Java, see the `How To's and Articles page
+<https://www.mongodb.com/developer/learn/?content=Articles&text=Java>`_.
```
Comment (ccho-mongodb) [inline source/index.txt:116]: Can you use this link instead? It uses the anchor target in the middle of the page where the Java results are rather than the top part of the page which is marketing material:
```suggestion
<https://www.mongodb.com/developer/learn/?content=Articles&text=java#main>`__.
```

File: source/index.txt [hunk 5]
```diff
@@ -66,9 +66,14 @@ core operations. This section contains information on how to:
...
-has detailed technical information about classes, methods, and
-configuration objects within the MongoDB Java driver.
+The MongoDB Java driver API documentation is split into several parts.
+Each part has detailed technical information about classes, methods, and
+configuration objects within the MongoDB Java driver:
+
+- `BSON <{+api+}/apidocs/bson/index.html>`__ (The BSON Layer)
+- `Core <{+api+}/apidocs/mongodb-driver-core/index.html>`__ (The Shared Core Classes)
+- `Java Driver <{+api+}/apidocs/mongodb-driver-sync/index.html>`__ (Modern API)
+- `Java Driver <{+api+}/apidocs/mongodb-driver-legacy/index.html>`__ (Legacy API)
```
Comment (ccho-mongodb) [inline source/index.txt:76]: 1) Not sure this formatting makes sense, especially using title case inside parentheses to describe each link. How about a list table instead with "Section" and "Description" as headings? Maybe you could describe them in a sentence rather than using short descriptions that sacrifice accuracy for shortness?

2) Is "layer" the correct terminology to use when describing the BSON library?

3) Just curious, but is the change that involves adding this to the "API Documentation" item in the left drawer part of this ticket? I think it would be great to include that change if it's not part of another ticket since we talked about it at the team PR review.

File: source/fundamentals.txt [hunk 6]
```diff
@@ -25,20 +25,20 @@ Fundamentals
    /fundamentals/csfle
    /fundamentals/time-series
 
-The Fundamentals section contains detailed information to help you
-expand your knowledge on how to use MongoDB with the Java driver. We
-divide the information in this section into the following topics: 
+The Fundamentals section is where you will find coverage of
+core operations. This section contains information on how to:
```
Comment (ccho-mongodb) [inline source/fundamentals.txt:29]: A couple items about this paragraph:

1) I the think "core" adjective in this context is used as an opinion. We should avoid opinion/editorial in our writing and try to keep it objective.
2) This list is introduced using a fragment rather thana sentence which we should avoid per the style guide: https://docs.mongodb.com/meta/style-guide/style/lists/#introductory-text

File: source/index.txt [hunk 7]
```diff
@@ -27,7 +27,7 @@ processing and reactive streams, we recommend the `Reactive Streams Driver
 
 The official MongoDB Java driver lets you connect your Java applications
 to MongoDB and work with your data. On this page, you can find descriptions of
-each section of the driver documentation.
+each section of the driver documentation and where to go for more resources.
```
Comment (ccho-mongodb) [inline source/index.txt:30]: I think "resources" doesn't help the reader understand what it is. See the other PR review comment on the "Resources" section.

File: source/index.txt [hunk 8]
```diff
@@ -95,11 +100,33 @@ What's New
...
+---------
+
+Visit the Developer Hub and MongoDB University for resources related to
+the MongoDB Java driver.
+
+Developer Hub
+~~~~~~~~~~~~~
+
+The Developer Hub contains content, resources, community, and social
+engagement for developers.
```
Comment (ccho-mongodb) [inline source/index.txt:113]: 1) I don't think the last two item of this list agrees with the verb "contains". 

2) It might be redundant to mention it contains content. Instead, why not mention what kind of content it contains (e.g. tutorials)?

File: source/index.txt [hunk 9]
```diff
@@ -95,11 +100,33 @@ What's New
...
+
+To view articles on Java, see the `How To's and Articles page
```
Comment (ccho-mongodb) [inline source/index.txt:115]: I think this info here and a few lines down (describing forums) could be a little more specific/precise. Pretend the user has to pay for every click on the page and therefore needs as much info as possible before deciding to visit a link. It might help to try to answer these questions:

1) Would you characterize the content of the articles as Java or Java Driver? 
2) What kind of articles are these?

Applies to the forums description as well.

File: source/index.txt [hunk 10]
```diff
@@ -95,11 +100,33 @@ What's New
...
+<https://www.mongodb.com/developer/learn/?content=Articles&text=Java>`_.
+
+To view the forum on Java, see the `forums page
+<https://www.mongodb.com/community/forums/tag/java>`_.
+
+MongoDB University
+~~~~~~~~~~~~~~~~~~
+
+MongoDB University contains free courses to inspire and educate everyone
+to learn MongoDB. It has the best-in-class training and empowers
```
Comment (ccho-mongodb) [inline source/index.txt:125]: Let's avoid opinionated descriptions like "inspire" and "best-in-class". Readers may lose confidence/trust when seeing opinionated material.

**Review by ccho-mongodb — CHANGES_REQUESTED** [review]
Good start! Left a couple comments on style issues and a few other recommendations.

File: source/index.txt (hunk 3 above)
Comment (kyuan-mongodb) [reply source/index.txt:103]: I changed it here because I saw it being more of a place with additional resources, but I agree, "Learn" is more accurate.

File: source/index.txt (hunk 5 above)
Comment (kyuan-mongodb) [reply source/index.txt:76]: 1. Agreed, I'll put this in a table
2. I took that wording from the GitHub pages site, but came across "Contains the base BSON classes." in the API docs, so I'll use that instead.
3. This was done previously in [this ticket](https://jira.mongodb.org/browse/DOCSP-17236)

File: source/index.txt [hunk 11]
```diff
@@ -95,11 +117,33 @@ What's New
...
+
+The Developer Hub provides tutorials and social engagement for
+developers.
+
+To learn how to use MongoDB features with the Java driver, see the `How
+To's and Articles page <https://www.mongodb.com/developer/learn/?content=Articles&text=java#main>`__.
+
+To learn what fellow developers are asking about, experiencing or their
+interests in regards to the Java driver, see the `forums page
+<https://www.mongodb.com/community/forums/tag/java>`__.
```
Comment (ccho-mongodb) [inline source/index.txt:137]: I like the new intro sentence in this sentence. It identifies two topics "tutorials" and "social engagement" followed by paragraphs that describe each of those. I see a couple potential improvements though:

1) I think we can connect the intro to these two paragraphs by either repeating the same or at least similar terms (tutorials and social engagement). Currently, you have "learn how" for "tutorials" and "learn what..." for social engagement. I think "learn how" might be ok for tutorials, but "learn what" doesn't relate that closely to social engagement. Maybe something like "ask questions and engage in discussions" could work.

2) The phrase "interests in regards to" is too wordy and seems indirect. Is there some alternative wording we can use to eliminate it?

File: source/index.txt [hunk 12]
```diff
@@ -95,11 +117,33 @@ What's New
...
+
+MongoDB University
+~~~~~~~~~~~~~~~~~~
+
+MongoDB University provides free courses to teach everyone how to use
+MongoDB.
 
 Take the free online course taught by MongoDB
```
Comment (ccho-mongodb) [inline source/index.txt:145]: I wonder if "taught by instructors at MongoDB" might be more accurate and informative.

File: source/index.txt [hunk 13]
```diff
@@ -66,9 +68,29 @@ core operations. This section contains information on how to:
 API Documentation
 -----------------
 
-The `API documentation <{+api+}/apidocs/mongodb-driver-sync/>`__
-has detailed technical information about classes, methods, and
-configuration objects within the MongoDB Java driver.
+The MongoDB Java driver API documentation is split into several sections.
+Each section has detailed technical information about classes, methods, and
+configuration objects within the MongoDB Java driver:
```
Comment (ccho-mongodb) [inline source/index.txt:73]: I think it's important to communicate that the API documentation is organized by package or library (contained in JARs). While it is true that it is split into sections, each section covers exactly one JAR and not parts of another JAR. Similarly, I would use package or library in the header of the table instead of "Section".

File: source/index.txt [hunk 14]
```diff
@@ -66,9 +68,29 @@ core operations. This section contains information on how to:
...
+.. list-table::
+   :header-rows: 1
+   :stub-columns: 1
+   :widths: 35 65
+
+   * - Section
+     - Description
+
+   * - `BSON <{+api+}/apidocs/bson/index.html>`__ 
+     - Contains the base BSON classes
```
Comment (ccho-mongodb) [inline source/index.txt:84]: Nitpick: you probably don't need "Contains the", especially if you change the header of this table from "Section" to library or package as requested in a comment further up.

**Review by ccho-mongodb — CHANGES_REQUESTED** [review]
Looks pretty good, just a few minor items to address. I would prefer to take a look at the revisions, so marking as changes requested.

**Review by ccho-mongodb — APPROVED** [review]
LGTM with the recommended sentences we discussed (after changing them to active voice).
